### ISSUE TITLE
Allow defining custom Firmware via datapacks

### ISSUE THREAD

**Issue by fnuecke** [fnuecke/oc2#127]
- Similar to how hard drives now work.
- The `file_system` dir name may either have to change, or will need a separate dir in parallel.

**Comment by atirutw** [comment]
I'd also like to be able to flash custom firmwares right in the game. This helps with developing and testing custom ones and thus running non-Linux stuff.

**Comment by samuelh2005** [comment]
Then one can use their favourite windows emulator for RISC-V!

**Comment by atirutw** [comment]
> Then one can use their favourite windows emulator for RISC-V!

I don't think there's a completely headless version let alone a RISC-V build of Windows

**Comment by samuelh2005** [comment]
> > Then one can use their favourite windows emulator for RISC-V!
> 
> 
> 
> I don't think there's a completely headless version let alone a RISC-V build of Windows

But people have got desktop version of Windows on a raspberry pi! If people can manage to do that then surely they can do the same for RISC-V

**Comment by CoolCat467** [comment]
That's totally different. A few builds of Windows exist for ARM64 processors, not RISC-V. It is not trivial to just change what architecture software runs for.

**Comment by walksanatora** [comment]
for me it is not windows or a diffrent linux
I want to make it so that I can "bare metal" program the robots

**Comment by atirutw** [comment]
Yeah. It doesn't really feel like OC if we can't get low and dirty down to bare metal like in OC1.

**Comment by walksanatora** [comment]
baremetal robots or bust :>
~~lua -> machine code compiler required to program them~~
